server/adaptors/zeppelin_backend.ts

```diff
@@ -0,0 +1,340 @@
+    } catch (error) {
+      throw new Error('Creating New Notebook Error');
+    }
+  };
+
+  // Rename a notebook in Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook/[noteId]/rename
+  renameNote = async function (
+    _context: RequestHandlerContext,
+    params: { name: string; noteId: string },
+    wreckOptions: optionsType
+  ) {
+    wreckOptions.payload = { name: params.name };
+    try {
+      const body = await requestor(
+        'PUT',
+        'api/notebook/' + params.noteId + '/rename/',
+        wreckOptions
+      );
+      return JSON.parse(body.toString());
+    } catch (error) {
+      throw new Error('Renaming Notebook Error');
+    }
+  };
+
+  // Clone a notebook in Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook/[noteId]
+  cloneNote = async function (
+    _context: RequestHandlerContext,
+    params: { name: string; noteId: string },
+    wreckOptions: optionsType
+  ) {
+    wreckOptions.payload = { name: params.name };
+    try {
+      const body = await requestor('POST', 'api/notebook/' + params.noteId, wreckOptions);
+      return JSON.parse(body.toString());
+    } catch (error) {
+      throw new Error('Cloning Notebook Error');
+    }
+  };
+
+  // Delete a notebook in Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook
+  deleteNote = async function (
+    _context: RequestHandlerContext,
+    noteId: string,
+    wreckOptions: optionsType
+  ) {
+    try {
+      const body = await requestor('DELETE', 'api/notebook/' + noteId, wreckOptions);
+      return JSON.parse(body.toString());
+    } catch (error) {
+      throw new Error('Deleting Notebook Error');
+    }
+  };
+
+  // Export a notebook from Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook/export/{noteid}
+  exportNote = async function (
+    _context: RequestHandlerContext,
+    noteId: string,
+    wreckOptions: optionsType
+  ) {
+    try {
+      const body = await requestor('GET', 'api/notebook/export/' + noteId, wreckOptions);
+      return JSON.parse(body.toString());
+    } catch (error) {
+      throw new Error('Export Notebook Error');
+    }
+  };
+
+  // Import a notebook in Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook/import
+  importNote = async function (
+    _context: RequestHandlerContext,
+    noteObj: any,
+    wreckOptions: optionsType
+  ) {
+    wreckOptions.payload = noteObj;
+    try {
+      const body = await requestor('POST', 'api/notebook/import', wreckOptions);
+      const respBody = JSON.parse(body.toString());
+      return respBody;
+    } catch (error) {
+      throw new Error('Import Notebook Error');
+    }
+  };
+
+  // Add a paragraph in notebook
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook/[noteId]/paragraph
+  addPara = async function (
```

Comment:
np: shortening Paragraph to Para is a bit confusing to me

Reply:
Changed and added documentation